templates/settings.py.erb

```diff
@@ -43,3 +43,10 @@ ALLOWED_CONTENT_CHECKSUMS = <%= scope['pulpcore::allowed_content_checksums'] %>
 
 # Derive HTTP/HTTPS via the X-Forwarded-Proto header set by Apache
 SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+
+CACHE_ENABLED = <%= scope['pulpcore::cache_enabled'] ? 'True' : 'False' %>
+<% if scope['pulpcore::cache_expires_ttl'] -%>
```

Comment:
```suggestion
<% if scope['pulpcore::cache_enabled'] && scope['pulpcore::cache_expires_ttl'] -%>
```

I assume including this setting when caching is disabled would not harm anything, but I think it would be better to omit unused settings from the configuration file.

Reply:
Looking at the code, it would appear there is no default here other than if this does not exist, then treat it as infinite cache -- https://github.com/pulp/pulpcore/commit/81684b206e3fd8bceb104e4a77c0d717387667f9#diff-a8f18c64735759c7f007d734b8b40e45938d03bc54327381fa2c9cf5fa2df544R42-R43

Then I would ask, is that what we want? Would that fill up Redis unknowingly?

Reply:
I think that code path would not be traversed with `CACHE_ENABLED = False`

Reply:
There are defaults for settings, so if you don't specify it, those will be used:
https://github.com/pulp/pulpcore/blob/9c1f115e9bc01e74381f09c5f3e28b06d2ae1ce1/pulpcore/app/settings.py#L242-L246

But @wbclark is right that if the cache is disabled, there really is no point in setting a TTL.

Reply:
It should use the default of 1 day if you don't set it explicitly. You might be interested in setting the `maxmemory` option for Redis just to make sure that Redis doesn't grow to eat up all the RAM (unlikely IMO since the entries will expire and are very small to begin with). https://redis.io/topics/lru-cache